plugins/inputs/beat/beat.go

```diff
@@ -78,10 +78,10 @@ type BeatStats struct {
 type Beat struct {
 	URL string `toml:"url"`
 
-	CollectBeatStats     bool `toml:"collect_beat_stats"`
-	CollectLibbeatStats  bool `toml:"collect_libbeat_stats"`
-	CollectSystemStats   bool `toml:"collect_system_stats"`
-	CollectFilebeatStats bool `toml:"collect_filebeat_stats"`
+	CollectBeatStats     *bool `toml:"collect_beat_stats"`
+	CollectLibbeatStats  *bool `toml:"collect_libbeat_stats"`
+	CollectSystemStats   *bool `toml:"collect_system_stats"`
+	CollectFilebeatStats *bool `toml:"collect_filebeat_stats"`
```

Comment:
Why do you need those to be pointers?

Reply:
Since we are setting defaults in `Init()` after the config has been read, it would not be possible to discern a false setting from an unset setting. Open to other ways to achieve this, this is how I've seen it done in other projects.

Reply:
set the defaults in init(), and then you don't have to check them at all, and the fields don't have to be pointers.

Reply:
@ssoroka This is how it was prior to the comments on https://github.com/influxdata/telegraf/pull/6653#discussion_r530877623

Reply:
Sorry about that. the correct way to set defaults is when you create the plugin, (around the lowercase `init()` call), By the time `Init()` is called, the config has been loaded over top of the defaults, and has the correct settings. Trying to set them in `Init()` leads to some weirdness, as you've experienced. I'd also like to go ahead with the change to change the bools to a `[]string` slice, because once we merge it the config is locked and we can't break backwards compatibility.

Reply:
@ssoroka this is my fault. I thought it is cleaner to initialize the defaults in `Init()` but completely missed that you cannot know if the boolean was specified as false by the user or it is false due to not being initialized. Again, my fault. Sorry @nferch for the confusion I caused!

Reply:
Ok, I've reverted the flow to set defaults in `init()`.